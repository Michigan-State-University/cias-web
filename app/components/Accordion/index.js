import React, { useEffect, memo } from 'react';
import PropTypes from 'prop-types';
import Reorder from 'react-reorder';

import { AccordionContainer } from './styled';
import Collapse from './Collapse';

const Accordion = ({
  children,
  onHide,
  onOpen,
  onReorder,
  opened,
  setOpened,
  onDelete,
}) => {
  useEffect(() => {
    if (opened !== -1) onOpen(opened);
  }, [opened]);

  const renderCollapse = (child, index) => {
    const { children: content, label, color } = child.props;

    const handleToggle = () => {
      let newIndex = index;
      if (opened === index) newIndex = -1;
      if (opened !== -1) onHide(opened);
      setOpened(newIndex);
    };

    const handleDelete = () => {
      onDelete(index);
    };

    return (
      <div key={`accordion-${index}`}>
        <Collapse
          onToggle={handleToggle}
          isOpened={opened === index}
          label={label}
          color={color}
          onDelete={handleDelete}
        >
          {content}
        </Collapse>
      </div>
    );
  };

  if (Array.isArray(children))
    return (
      <AccordionContainer>
        <Reorder
          holdTime={125}
          reorderId="blocks-list"
          onReorder={onReorder}
          disabled={!onReorder}
        >
          {children.map(renderCollapse)}
        </Reorder>
      </AccordionContainer>
    );

  return <AccordionContainer>{renderCollapse(children, 0)}</AccordionContainer>;
};

Accordion.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
  onHide: PropTypes.func,
  onOpen: PropTypes.func,
  onReorder: PropTypes.func,
  opened: PropTypes.number,
  setOpened: PropTypes.func,
  onDelete: PropTypes.func,
};

Accordion.defaultProps = {};

export default memo(Accordion);
