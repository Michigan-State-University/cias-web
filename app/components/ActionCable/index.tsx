import { ActionCableProvider, ActionCableContext } from './ActionCableProvider';

export { ActionCableProvider, ActionCableContext };
