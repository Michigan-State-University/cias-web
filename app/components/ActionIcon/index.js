/**
 *
 * ActionIcon
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon';

import cross from 'assets/svg/cross.svg';
import { CrossLink, CrossButton } from './styled';

const ActionIcon = ({ to, onClick, iconSrc, ...restProps }) => {
  const icon = <Icon src={iconSrc} alt="cross" />;
  if (to)
    return (
      <CrossLink data-cy="back-intervention-button" to={to}>
        {icon}
      </CrossLink>
    );
  if (onClick)
    return (
      <CrossButton onClick={onClick} {...restProps}>
        {icon}
      </CrossButton>
    );
  return null;
};

ActionIcon.propTypes = {
  to: PropTypes.string,
  onClick: PropTypes.func,
  iconSrc: PropTypes.any,
};

ActionIcon.defaultProps = {
  iconSrc: cross,
};

export default ActionIcon;
