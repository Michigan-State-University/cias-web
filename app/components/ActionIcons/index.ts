export * from './PlayStopButton';
export * from './ClearFiltersButton';
