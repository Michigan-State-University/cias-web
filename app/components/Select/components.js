import React from 'react';
import get from 'lodash/get';
import { components } from 'react-select';

import Checkbox from 'components/Checkbox';
import ToggleArrow from 'components/ToggleArrow';

import { OptionContainer } from './styled';

const DropdownIndicator = (props) => {
  const isOpen = get(props, 'selectProps.menuIsOpen', false);
  return (
    <components.DropdownIndicator {...props}>
      <ToggleArrow facingUp={isOpen} />
    </components.DropdownIndicator>
  );
};

DropdownIndicator.propTypes = components.DropdownIndicator.propTypes;

const Option = ({
  innerProps,
  innerRef,
  data,
  isSelected,
  isFocused,
  selectProps,
}) => {
  const { label, additionalData } = data;
  const { formatLabel } = selectProps;
  const getLabel = () => {
    if (!additionalData) return formatLabel(label);
    return formatLabel(label, additionalData);
  };

  /*
      Make react-select not propagate click event up the DOM as it breaks outside click handler
   */
  const onClick = (e) => {
    e.nativeEvent.stopImmediatePropagation();
    innerProps.onClick(e);
  };

  const modifiedInnerProps = { ...innerProps, onClick };

  return (
    <OptionContainer
      ref={innerRef}
      {...modifiedInnerProps}
      isFocused={isFocused}
      className="custom-option"
    >
      <Checkbox id={label} checked={isSelected} mr={15}>
        {getLabel()}
      </Checkbox>
    </OptionContainer>
  );
};

Option.propTypes = components.Option.propTypes;

const DefaultOption = ({ innerProps, ...props }) => {
  /*
      Make react-select not propagate click event up the DOM as it breaks outside click handler
   */
  const onClick = (e) => {
    e.nativeEvent.stopImmediatePropagation();
    innerProps.onClick(e);
  };

  // eslint-disable-next-line no-param-reassign
  props.innerProps = { ...innerProps, onClick };
  return <components.Option {...props} />;
};

DefaultOption.propTypes = components.Option.propTypes;

export { DropdownIndicator, Option, DefaultOption };
