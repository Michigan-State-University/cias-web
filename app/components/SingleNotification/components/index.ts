export * from './NewConversationNotificationLayout';
export * from './NewNarratorWasSetNotificationLayout';
