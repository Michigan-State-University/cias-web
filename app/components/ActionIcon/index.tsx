/**
 *
 * ActionIcon
 *
 */

import React, { MouseEventHandler, PropsWithChildren } from 'react';
import { useIntl } from 'react-intl';

import Icon from 'components/Icon';
import { LayoutProps, MarginProps } from 'components/BaseComponentStyles';

import cross from 'assets/svg/cross.svg';

import { CrossLink, CrossButton } from './styled';
import messages from './messages';

type Props = PropsWithChildren<{
  to?: string;
  onClick?: MouseEventHandler;
  iconSrc: SVGElement;
  ariaText: string;
}> &
  LayoutProps &
  MarginProps &
  Record<string, unknown>;

const ActionIcon = ({
  to,
  onClick,
  iconSrc = cross,
  ariaText,
  children,
  ...restProps
}: Props) => {
  const { formatMessage } = useIntl();

  const icon = <Icon src={iconSrc} alt={ariaText} />;
  if (to)
    return (
      <CrossLink
        data-cy="back-intervention-button"
        to={to}
        title={ariaText ?? formatMessage(messages.defaultCrossButtonText)}
        aria-label={ariaText ?? formatMessage(messages.defaultCrossButtonText)}
        {...restProps}
      >
        {icon}
        {children}
      </CrossLink>
    );
  if (onClick)
    return (
      <CrossButton
        onClick={onClick}
        title={ariaText ?? formatMessage(messages.defaultCrossButtonText)}
        aria-label={ariaText ?? formatMessage(messages.defaultCrossButtonText)}
        {...restProps}
      >
        {icon}
        {children}
      </CrossButton>
    );
  return null;
};

export default ActionIcon;
