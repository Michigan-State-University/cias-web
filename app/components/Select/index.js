/**
 *
 * Select
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import Box from 'components/Box';
import { themeColors } from 'theme';

import ReactSelect from 'react-select';
import { DropdownIndicator, Option } from './components';

const customStyles = ({ isMulti, bg, isDisabled }) => ({
  control: provided => ({
    ...provided,
    borderWidth: '1px',
    borderRadius: '5px',
    borderColor: `${themeColors.highlight}`,
    boxShadow: '0',
    height: isMulti ? 'auto' : '45px',
    minHeight: '45px',
    width: '100%',
    background: `${bg || 'auto'}`,
    '&:hover': {
      borderColor: `${themeColors.highlight}`,
    },
    cursor: isDisabled ? 'not-allowed' : 'pointer',
  }),
  option: provided => ({
    ...provided,
    cursor: isDisabled ? 'not-allowed' : 'pointer',
  }),
  menuPortal: provided => ({ ...provided, zIndex: 999 }),
  placeholder: provided => ({
    ...provided,
    color: 'hsl(0, 0%, 40%)',
  }),
});

const customComponents = isMulti => ({
  IndicatorSeparator: () => null,
  DropdownIndicator: props => <DropdownIndicator {...props} />,
  LoadingIndicator: () => null,
  ...(isMulti
    ? {
        Option: props => <Option {...props} />,
      }
    : {}),
});

const Select = ({ selectProps, ...restProps }) => (
  <Box {...restProps}>
    <ReactSelect
      components={customComponents(selectProps.isMulti)}
      menuPortalTarget={document.body}
      styles={customStyles(selectProps)}
      {...selectProps}
    />
  </Box>
);

Select.propTypes = {
  selectProps: PropTypes.object,
  bg: PropTypes.string,
};

export default Select;
