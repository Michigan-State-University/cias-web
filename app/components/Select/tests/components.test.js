/**
 *
 * Tests for components
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Option } from '../components';

const defaultProps = {
  data: { label: 'test' },
  isSelected: true,
  isFocused: true,
  selectProps: { formatLabel: jest.fn(s => s) },
};

describe('<Option />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Option {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(<Option {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
  it('Should render and match the snapshot', () => {
    const newProps = {
      data: { label: 'test', additionalData: '2' },
      isSelected: true,
      isFocused: true,
      selectProps: { formatLabel: jest.fn((s, t) => `${s}${t}`) },
    };
    const { container } = render(<Option {...newProps} />);
    expect(container).toMatchSnapshot();
  });
});
