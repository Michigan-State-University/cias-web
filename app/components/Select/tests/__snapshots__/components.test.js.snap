// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option /> Should render and match the snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  justify-items: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: none;
}

.c7 {
  margin-right: 5px;
  border-radius: 5px;
}

.c2:disabled + label > .c6 {
  opacity: 0.5;
  cursor: not-allowed;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c5:hover {
  cursor: pointer;
}

.c2:disabled + .c4 {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0 {
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #E9F1FB;
  cursor: pointer;
}

<div>
  <div
    class="c0 custom-option"
  >
    <div
      class="c1"
    >
      <input
        checked=""
        class="c2 c3"
        id="test"
        type="checkbox"
      />
      <label
        class="c4 c5"
        for="test"
      >
        <img
          class="c6 c7"
          src="IMAGE_MOCK"
        />
        test
      </label>
    </div>
  </div>
</div>
`;

exports[`<Option /> Should render and match the snapshot 2`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  justify-items: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: none;
}

.c7 {
  margin-right: 5px;
  border-radius: 5px;
}

.c2:disabled + label > .c6 {
  opacity: 0.5;
  cursor: not-allowed;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c5:hover {
  cursor: pointer;
}

.c2:disabled + .c4 {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0 {
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #E9F1FB;
  cursor: pointer;
}

<div>
  <div
    class="c0 custom-option"
  >
    <div
      class="c1"
    >
      <input
        checked=""
        class="c2 c3"
        id="test"
        type="checkbox"
      />
      <label
        class="c4 c5"
        for="test"
      >
        <img
          class="c6 c7"
          src="IMAGE_MOCK"
        />
        test2
      </label>
    </div>
  </div>
</div>
`;
